pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -3608,6 +3652,67 @@ func pruneRegistrationEntries(tx *gorm.DB, expiresBefore time.Time, logger logru
 	return nil
 }
 
+func createRegistrationEntryEvent(tx *gorm.DB, entryID string) error {
+	newRegisteredEntryEvent := RegisteredEntryEvent{
+		EntryID: entryID,
+	}
+
+	if err := tx.Create(&newRegisteredEntryEvent).Error; err != nil {
+		return sqlError.Wrap(err)
+	}
+
+	return nil
+}
+
+func listRegistrationEntriesEvents(tx *gorm.DB, req *datastore.ListRegistrationEntriesEventsRequest) (*datastore.ListRegistrationEntriesEventsResponse, error) {
+	if req.Pagination != nil && req.Pagination.PageSize == 0 {
+		return nil, status.Error(codes.InvalidArgument, "cannot paginate with pagesize = 0")
+	}
+
+	p := req.Pagination
+	var err error
+	if p != nil {
+		tx, err = applyPagination(p, tx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var events []RegisteredEntryEvent
+	if err := tx.Find(&events, "id > ?", req.GreaterThanEventID).Error; err != nil {
+		return nil, sqlError.Wrap(err)
+	}
+
+	if p != nil {
+		p.Token = ""
+		// Set token only if page size is the same as events len
+		if len(events) > 0 {
+			lastEntry := events[len(events)-1]
+			p.Token = fmt.Sprint(lastEntry.ID)
+		}
+	}
+
+	resp := &datastore.ListRegistrationEntriesEventsResponse{
+		Pagination: p,
+	}
+	for _, event := range events {
+		resp.EntryIDs = append(resp.EntryIDs, event.EntryID)
+	}
+	if len(events) > 0 {
+		resp.FirstEventID = events[0].ID
+	}
+
+	return resp, nil
+}
+
+func pruneRegistrationEntriesEvents(tx *gorm.DB, olderThan time.Duration) error {
+	if err := tx.Where("created_at < DATE_SUB(NOW(), INTERVAL ? SECOND)", olderThan.Seconds()).Delete(&RegisteredEntryEvent{}).Error; err != nil {
```

Comment:
Does this work on all SQL variants we support? I don't think PostgreSQL has DATE_SUB....

Reply:
Removed date_sub

Reply:
tested with all supported sql variants.